parser/src/lexer.rs

```diff
@@ -1105,6 +1105,84 @@ where
         let tok_end = self.get_pos();
         Ok((tok_start, ty, tok_end))
     }
+
+    fn lex_byte(&self, s: String) -> Result<Vec<u8>, LexicalError> {
+        let mut res = vec![];
+        let mut escape = false; //flag if previous was \
+        let mut hex_on = false; // hex mode on or off
+        let mut hex_value = String::new();
+
+        for c in s.chars() {
+            match c {
+                '\\' => {
+                    if escape {
+                        res.push(92);
+                        escape = false;
+                    } else {
+                        escape = true;
+                    }
+                }
+
+                'x' => {
+                    if escape {
+                        hex_on = true;
+                    } else {
+                        res.push(120);
+                    }
+                    escape = false;
+                }
+                't' => {
+                    if escape {
+                        res.push(9);
+                    } else {
+                        res.push(116);
+                    }
+                    escape = false;
+                }
+                'n' => {
+                    if escape {
+                        res.push(10);
+                    } else {
+                        res.push(110)
+                    }
+                    escape = false;
+                }
+                'r' => {
+                    if escape {
+                        res.push(13);
+                    } else {
+                        res.push(114)
+                    }
+                    escape = false;
+                }
+                x => {
+                    if hex_on {
+                        if x.is_ascii_hexdigit() {
+                            if hex_value.is_empty() {
+                                hex_value.push(x);
+                                continue;
+                            } else {
+                                hex_value.push(x);
+                                res.push(u8::from_str_radix(&hex_value, 16).unwrap());
+                                hex_on = false;
+                                hex_value.clear();
+                            }
+                        } else {
+                            return Err(LexicalError::StringError);
+                        }
+                    } else {
+                        if escape {
+                            res.push(92);
```

Comment:
Always push 92 here? Not `c`?

Reply:
yes backslash (92) was seen at previous loop but not added since you need the following char to know if its a "pure" backslash or a backslash of a \xXX. So I add the "pure" backslash at next loop before to add `c` if we are not `hex_on`

Reply:
Maybe you could do `'\\' as u8` to make it more clear what character you're adding.

Reply:
I thought everyone knows ascii table by heart :grin:

Reply:
following clippy i'll use b'x' notation : https://rust-lang.github.io/rust-clippy/master/index.html#char_lit_as_u8